src/hermit.rs

```diff
@@ -43,11 +43,38 @@ impl<T: Config> Hermit<T> {
         let path = self.config.shell_root_path().join(name);
         file_ops.create_git_repo(path);
     }
+
+    pub fn inhabit_shell(&mut self, file_ops: &mut FileOperations, name: &str) -> Result<Vec<String>, Error> {
+        let mut shell_names = Vec::new();
+        let mut res = Ok(());
+        if name == "" {
+            shell_names = self.get_shell_list();
+        } else {
+            res = self.set_current_shell(name);
+        }
```

Comment:
I think this `if` statement should go in `handle_inhabit_subcommand` only.  Currently you have the same if statement in two places.